client/send.go

```diff
@@ -244,8 +245,9 @@ func (s *Session) BlockingSendUnreliableMessage(recipient, provider string, mess
 	select {
 	case reply := <-replyWaitChan:
 		return reply, nil
-	// these timeouts are often far too aggressive
-	case <-time.After(sentMessage.ReplyETA + cConstants.RoundTripTimeSlop):
+	case <-s.HaltCh():
+		return nil, ErrHalted
+	case <-time.After(sentMessage.ReplyETA + sentMessage.ReplyETA>>1):
```

Comment:
Using >> or << is far less readable than / or *. It also doesn't do what you think unless they are unsigned ints... furthermore regardless of the operator used here for division, there needs to be parentheses so that the reviewer doesn't need to go look up the golang specification and read the operator precedence ordering which is:

Precedence    Operator
    5             *  /  %  <<  >>  &  &^
    4             +  -  |  ^
    3             ==  !=  <  <=  >  >=
    2             &&
    1             ||

Reply:
d04ecb98f04d5cca08da1d2a9e00efe07679b1e2 f7427a08fa76fad1243d5fe9c76df1c8ef413262 change the timeout behavior to use the RoundTripTimeSlop, reduces RoundTripTimeSlop to 4 seconds, and changes the clients garbage collector behavior so that kaetzchen responders can respond within 2 epochs